Tests/SentryTests/Integrations/OutOfMemory/SentryOutOfMemoryScopeObserverTests.swift

```diff
@@ -0,0 +1,131 @@
+import XCTest
+
+class SentryOutOfMemoryScopeObserverTests: XCTestCase {
+    private class Fixture {
+        let breadcrumb: Breadcrumb
+        let options: Options
+        let fileManager: SentryFileManager
+        let currentDate = TestCurrentDateProvider()
+
+        init() {
+            breadcrumb = Breadcrumb()
+            breadcrumb.level = SentryLevel.info
+            breadcrumb.timestamp = Date(timeIntervalSince1970: 10)
+            breadcrumb.category = "category"
+            breadcrumb.type = "user"
+            breadcrumb.message = "Click something"
+
+            options = Options()
+            fileManager = try! SentryFileManager(options: options, andCurrentDateProvider: currentDate)
+        }
+
+        func getSut() -> SentryOutOfMemoryScopeObserver {
+            return getSut(fileManager: self.fileManager)
+        }
+
+        func getSut(fileManager: SentryFileManager) -> SentryOutOfMemoryScopeObserver {
+            return SentryOutOfMemoryScopeObserver(maxBreadcrumbs: 10, fileManager: fileManager)
+        }
+    }
+
+    private var fixture: Fixture!
+    private var sut: SentryOutOfMemoryScopeObserver!
+
+    override func setUp() {
+        super.setUp()
+
+        fixture = Fixture()
+        sut = fixture.getSut()
+    }
+
+    override func tearDown() {
+        super.tearDown()
+        fixture.fileManager.deleteAllFolders()
+    }
+
+    // Test that we're storing the serialized breadcrumb in a proper JSON string
+    func testStoreBreadcrumb() throws {
+        let breadcrumb = fixture.breadcrumb.serialize() as! [String: String]
+
+        sut.addSerializedBreadcrumb(breadcrumb)
+
+        let fileOneContents = try String(contentsOfFile: fixture.fileManager.breadcrumbsFilePathOne)
+        let firstLine = String(fileOneContents.split(separator: "\n").first!)
+        let dict = try JSONSerialization.jsonObject(with: firstLine.data(using: .utf8)!) as! [String: String]
+
+        XCTAssertEqual(dict, breadcrumb)
+    }
+
+    func testStoreInMultipleFiles() throws {
+        let breadcrumb = fixture.breadcrumb.serialize()
+
+        var count = 0
+        while count < 9 {
+            sut.addSerializedBreadcrumb(breadcrumb)
+            count += 1
+        }
```

Comment:
`l`: A bit shorter if you want.
```suggestion
        for _ in 0..<count {
            sut.addSerializedBreadcrumb(breadcrumb)
        }
```